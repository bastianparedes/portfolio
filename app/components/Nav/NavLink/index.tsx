import React from 'react';

import styles from './styles.module.scss';

interface NavLinkProps {
  link: string;
  title: string;
  closeNav: () => void;
}

const NavLink = ({ link, title, closeNav }: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a className={styles.link__a} href={'#' + link} onClick={closeNav}>
        {title}
      </a>
    </li>
  );
};

export default NavLink;
