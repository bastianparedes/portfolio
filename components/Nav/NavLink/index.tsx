import React, { MouseEventHandler } from 'react'
import styles from './styles.module.css'

interface NavLinkProps {
  link: string
  title: string
  onClick: MouseEventHandler<HTMLAnchorElement>
}

const NavLink = (props: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        href={'#' + props.link}
        className={styles.link__a}
        onClick={props.onClick}
      >
        {props.title}
      </a>
    </li>
  )
}

export default NavLink
