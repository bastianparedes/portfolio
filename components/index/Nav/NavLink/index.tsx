import React from 'react';

import styles from './styles.module.scss';

interface NavLinkProps {
  link: string;
  title: string;
  closeNav: () => void;
}

const NavLink = ({ link, title, closeNav }: NavLinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a href={'#' + link} className={styles.link__a} onClick={closeNav}>
        {title}
      </a>
    </li>
  );
};

export default NavLink;
