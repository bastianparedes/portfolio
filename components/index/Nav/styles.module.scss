@import 'bastianparedes/styles/_breakpoints';

.nav {
  height: fit-content;
  background-color: rgb(0, 16, 85);
  width: 100%;
  position: sticky;
  top: 0;
  display: grid;
  grid-template-columns: 1fr 1fr;
  z-index: 3;
}

.navLeft {
  grid-column: 1;
  grid-row: 1;
  height: 4rem;
  position: relative;
  object-fit: cover;
  display: flex;
  align-items: center;
}

.logoContainer {
  width: 116.63px;
  height: 51.2px;
  position: relative;
  margin-left: 1rem;
}

.buttonContainer {
  grid-column: 2;
  grid-row: 1;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.button {
  width: 3.5rem;
  height: 3.5rem;
  font-size: 2.7rem;
  display: none;
  align-items: center;
  justify-content: center;
  border: none;
  margin-right: 1rem;
  cursor: pointer;
  transition: all ease 1s;
  z-index: 1;
}

.buttonAnimation {
  transform: rotate(180deg);
}

.linksContainer {
  grid-column: 2;
  grid-row: 1;
  width: fit-content;
  list-style: none;
  margin: 0 3rem 0 auto;
  display: flex;
  align-items: center;
  gap: 3rem;
  font-size: 2rem;
  background-color: inherit;
}

@include phone-and-tablet {
  .button {
    display: flex;
  }

  .linksContainer {
    width: 100%;
    grid-column: 1 / 3;
    grid-row: 2;
    position: absolute;
    display: grid;
    justify-content: center;
    align-items: center;
    grid-template-columns: 1fr;
    grid-auto-rows: 1fr;
    gap: 0;
    margin: 0rem 0;
    padding: 0;
    border-color: rgb(133, 133, 133);
    border-width: 3px;
    border-style: solid none;
    transition: all linear 0.5s;
    transform: scaleY(0);
    transform-origin: top;
  }

  .linksContainerOpened {
    transform: scaleY(1);
  }
}
