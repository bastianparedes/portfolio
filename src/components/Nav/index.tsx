import React from 'react'
import { AiOutlineMenu } from 'react-icons/ai'
import styles from './styles.module.css'

interface LinkProps {
  link: string
  title: string
}

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false)
  const logoPath = process.env.PUBLIC_URL + '/images/logo.png'
  const linkInfo = [
    {
      link: '#home',
      title: 'Home'
    },
    {
      link: '#about',
      title: 'About'
    },
    {
      link: '#portfolio',
      title: 'Porftolio'
    }
  ]
  const buttonClickHandler = (): void => {
    setOpened((value) => !value)
  }

  const openNav = (): void => {
    setOpened(false)
  }

  const Link = (props: LinkProps): JSX.Element => {
    return (
      <li className={styles.link}>
        <a href={props.link} className={styles.link__a} onClick={openNav}>
          {props.title}
        </a>
      </li>
    )
  }

  return (
    <nav className={styles.nav}>
      <div className={styles.logoContainer}>
        <img className={styles.logo} src={logoPath} alt="BastiÃ¡n Paredes" />
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={`${styles.button} ${opened ? styles.buttonAnimation : ''}`}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={`${styles.linksContainer} ${
          opened ? styles.linksContainerOpened : ''
        }`}
      >
        {linkInfo.map((info) => (
          <Link key={info.link} {...info} />
        ))}
      </ul>
    </nav>
  )
}

export default Nav
