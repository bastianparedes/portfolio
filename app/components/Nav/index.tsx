'use client';

import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/image';
import { AiOutlineMenu } from 'react-icons/ai';

import NavLink from './NavLink';
import styles from './styles.module.scss';
import constants from '../../../config/constants';
import logo from '../../../public/images/logo.png';

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false);

  const buttonClickHandler = (): void => {
    setOpened((value) => !value);
  };

  const closeNav = (): void => {
    setOpened(false);
  };

  return (
    <nav className={styles.nav}>
      <div className={styles.navLeft}>
        <div className={styles.logoContainer}>
          <Image
            alt={'BastiÃ¡n Paredes Logo'}
            placeholder="blur"
            src={logo}
            style={{
              height: '100%',
              objectFit: 'contain',
              width: '100%'
            }}
          />
        </div>
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={classNames(
            styles.button,
            opened && styles.buttonAnimation
          )}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={classNames(
          styles.linksContainer,
          opened && styles.linksContainerOpened
        )}
        data-testid="data-testid-nav__ul"
      >
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.HOME.ID}
          title={constants.NAV.BUTTONS.HOME.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.ABOUT.ID}
          title={constants.NAV.BUTTONS.ABOUT.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.PORTFOLIO.ID}
          title={constants.NAV.BUTTONS.PORTFOLIO.TITLE}
        />
      </ul>
    </nav>
  );
};

export default Nav;
