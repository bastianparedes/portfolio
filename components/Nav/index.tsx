import React from 'react'
import { AiOutlineMenu } from 'react-icons/ai'
import styles from './styles.module.css'
import constants from '../../config/constants.json'
import Image from 'next/image'
import NavLink from './NavLink'
import classNames from '../../utils/classNames'

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false)

  const buttonClickHandler = (): void => {
    setOpened((value) => !value)
  }

  const closeNav = (): void => {
    setOpened(false)
  }

  return (
    <nav className={styles.nav}>
      <div className={styles.navLeft}>
        <div className={styles.logoContainer}>
          <Image
            src={'/images/logo.png'}
            alt="BastiÃ¡n Paredes Logo"
            layout="fill"
            objectFit="contain"
          />
        </div>
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={classNames(
            styles.button,
            opened && styles.buttonAnimation
          )}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={classNames(
          styles.linksContainer,
          opened && styles.linksContainerOpened
        )}
      >
        <NavLink
          link={constants.NAV.BUTTONS.HOME.ID}
          title={constants.NAV.BUTTONS.HOME.TITLE}
          onClick={closeNav}
        />
        <NavLink
          link={constants.NAV.BUTTONS.ABOUT.ID}
          title={constants.NAV.BUTTONS.ABOUT.TITLE}
          onClick={closeNav}
        />
        <NavLink
          link={constants.NAV.BUTTONS.PORTFOLIO.ID}
          title={constants.NAV.BUTTONS.PORTFOLIO.TITLE}
          onClick={closeNav}
        />
      </ul>
    </nav>
  )
}

export default Nav
