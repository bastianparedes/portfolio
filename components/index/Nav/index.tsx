import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/legacy/image';
import { AiOutlineMenu } from 'react-icons/ai';

import NavLink from './NavLink';
import styles from './styles.module.scss';
import constants from '../../../config/constants';

const Nav = (): JSX.Element => {
  const [opened, setOpened] = React.useState(false);

  const buttonClickHandler = (): void => {
    setOpened((value) => !value);
  };

  const closeNav = (): void => {
    setOpened(false);
  };

  return (
    <nav className={styles.nav}>
      <div className={styles.navLeft}>
        <div className={styles.logoContainer}>
          <Image
            alt="BastiÃ¡n Paredes Logo"
            layout="fill"
            objectFit="contain"
            placeholder="blur"
            src="/images/logo.png"
          />
        </div>
      </div>
      <div className={styles.buttonContainer}>
        <button
          className={classNames(
            styles.button,
            opened && styles.buttonAnimation
          )}
          data-testid="data-testid-nav__button"
          onClick={buttonClickHandler}
        >
          <AiOutlineMenu />
        </button>
      </div>
      <ul
        className={classNames(
          styles.linksContainer,
          opened && styles.linksContainerOpened
        )}
        data-testid="data-testid-nav__ul"
      >
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.HOME.ID}
          title={constants.NAV.BUTTONS.HOME.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.ABOUT.ID}
          title={constants.NAV.BUTTONS.ABOUT.TITLE}
        />
        <NavLink
          closeNav={closeNav}
          link={constants.NAV.BUTTONS.PORTFOLIO.ID}
          title={constants.NAV.BUTTONS.PORTFOLIO.TITLE}
        />
      </ul>
    </nav>
  );
};

export default Nav;
