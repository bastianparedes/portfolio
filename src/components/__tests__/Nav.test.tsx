import React from 'react'
import Nav from '../Nav'
import { render, screen, fireEvent } from '@testing-library/react'

describe('<Nav />', () => {
  it('should render', () => {
    const { container } = render(<Nav />)
    expect(container).toMatchSnapshot()
  })

  it('should open nav and closses when link clicked', () => {
    render(<Nav />)
    const button = screen.getByTestId('data-testid-nav__button')
    const link = screen.getByText('Home')
    fireEvent.click(link)
    fireEvent.click(button)
    expect(screen).toMatchSnapshot()
  })
})
