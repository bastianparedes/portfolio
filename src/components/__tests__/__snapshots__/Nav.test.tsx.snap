// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Nav /> should open nav and closses when link clicked 1'] = `
Object {
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "logTestingPlaygroundURL": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
}
`;

exports['<Nav /> should render 1'] = `
<div>
  <nav
    class="nav"
  >
    <div
      class="logoContainer"
    >
      <img
        alt="BastiÃ¡n Paredes"
        class="logo"
        src="/images/logo.png"
      />
    </div>
    <div
      class="buttonContainer"
    >
      <button
        class="button "
        data-testid="data-testid-nav__button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </button>
    </div>
    <ul
      class="linksContainer "
    >
      <li
        class="link"
      >
        <a
          class="link__a"
          href="#home"
        >
          Home
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="#about"
        >
          About
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="link__a"
          href="#portfolio"
        >
          Porftolio
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
